util/object_pool.c

```diff
@@ -655,6 +671,29 @@ static void *eviction_thread(void *arg)
     return NULL;
 }
 
+static int objpool_signal_unexhausted_int(comdb2_objpool_t op)
+{
+    assert(op != NULL);
+    OP_DBG(op, "signal unexhausted");
+    Pthread_cond_signal(&op->unexhausted);
+    return 0;
+}
+
+static int objpool_notify_int(comdb2_objpool_t op, int force)
+{
+    int rc;
+    assert(op != NULL);
+    OP_DBG(op, "notify pool");
+    Pthread_mutex_lock(&op->data_mutex);
+    if (force || (op->nborrowwaits != 0)) {
```

Comment:
Only signal when there's waiters (ie, op->nborrowwaits != 0).

Reply:
Yeah, I agree with that.  I was just originally nervous about relying on nborrowwaits...  It seems there wouldn't be real harm either way?